samples/sensor/stm32_temp_sensor/sample.yaml

```diff
@@ -0,0 +1,10 @@
+sample:
+  description: Usage of STM32 temperature sensor
+  name: stm32_temp_sensor
+tests:
+  sample.sensor.stm32_temp_sensor:
+    platform_allow: nucleo_f401re
+    depends_on: adc
+    tags: sensors
+    build_only: true
+    filter: dt_compat_enabled("st,stm32-temp")
```

Comment:
I think the reasoning to define filter should be:
`What is the prerequisite for a board to be compatible with this sample ?`

On my side, I'll try `dt_compat_enabled_with_label("st,stm32-adc", "ADC_1")`.
I haven't tested, so please have a try.
If not working, let's forget about the filter and stick to `platform_allow`

Reply:
@erwango I'm not sure about proposed filter. It should build a sample with any STM32 device with ADC enabled.
Maybe, there is any solution to add .overlay when CONFIG_STM32_TEMP enabled? And this allows all builds with CONFIG_SOC_FAMILY_STM32=y